### PR TITLE
Fix for TLS1.3 handshake failure 

### PR DESCRIPTION
**Pull request by sahilnxp**
Fix for TLS1.3 handshake failure.

### CODE REVIEW

**Comment by embetrix** [comment]
Hi @sahilnxp 

thanks for the Fix :-)

I will give it a try when I'm back from Holidays on 22/01/2024

File: core/drivers/crypto/caam/acipher/caam_ecc.c [hunk 1]
```diff
@@ -477,6 +564,7 @@ static TEE_Result do_verify(struct drvcrypt_sign_data *sdata)
 	struct caamdmaobj sign_c = { };
 	struct caamdmaobj sign_d = { };
 	uint32_t pdb_sgt_flags = 0;
+	struct caambuf caambuf_msg = {};
```
Comment (etienne-lms) [inline core/drivers/crypto/caam/acipher/caam_ecc.c:567]: add  aspace char in between the braces

File: core/drivers/crypto/caam/crypto.mk [hunk 2]
```diff
@@ -83,6 +83,11 @@ $(call force, CFG_JR_INDEX,2)
 $(call force, CFG_JR_INT,146)
 $(call force, CFG_NXP_CAAM_SGT_V1,y)
 $(call force, CFG_JR_HAB_INDEX,0)
+# There is a limitation on i.MX8M platforms wrt to ECDSA Signing/Verification
```
Comment (etienne-lms) [inline core/drivers/crypto/caam/crypto.mk:86]: prefer s/wrt/regarding/ ?

File: core/drivers/crypto/caam/crypto.mk [hunk 3]
```diff
@@ -83,6 +83,11 @@ $(call force, CFG_JR_INDEX,2)
...
+# Size of Class 2 Context register is 40bytes, because of which sigin/verify
```
Comment (etienne-lms) [inline core/drivers/crypto/caam/crypto.mk:87]: s/sigin/sign/

**Review by etienne-lms — COMMENTED** [review]
Commit message for commit "":
```patch
-Fixes: 6492
+Fixes: https://github.com/OP-TEE/optee_os/issues/6492

**Comment by clementfaure** [comment]
With the comments from @etienne-lms addressed,
`Acked-by: Clement Faure <clement.faure@nxp.com>`

**Comment by sahilnxp** [comment]
> ```diff
>  https://github.com/OP-TEE/optee_os/issues/6492
> ```

Checkpatch gives warning on this:
```
#/linux-factory/optee-os$ scripts/checkpatch.pl 0002-drivers-caam-Fix-for-TLS1.3-handshake-failure.patch
WARNING: Unknown link reference 'Fixes:', use 'Link:' or 'Closes:' instead
#24:
Fixes: https://github.com/OP-TEE/optee_os/issues/6492

total: 0 errors, 1 warnings, 0 checks, 224 lines checked
#/linux-factory/optee-os$
```

**Comment by sahilnxp** [comment]
> Commit message for commit "":
> 
> ```diff
> -Fixes: 6492
> +Fixes: https://github.com/OP-TEE/optee_os/issues/6492
> ```

For this Checkpatch gives a warning: https://github.com/OP-TEE/optee_os/actions/runs/7470932819/job/20330388206?pr=6583#step:5:37

**Comment by jforissier** [comment]
> > Commit message for commit "":
> > ```diff
> > -Fixes: 6492
> > +Fixes: https://github.com/OP-TEE/optee_os/issues/6492
> > ```
> 
> For this Checkpatch gives a warning: https://github.com/OP-TEE/optee_os/actions/runs/7470932819/job/20330388206?pr=6583#step:5:37

`Fixes:` should be followed by a commit ID. Use `Link:` here.

**Comment by sahilnxp** [comment]
Pull request updated and tags applied.

**Comment by jenswi-linaro** [comment]
Since it fixes a problem, how about adding a `Fixes:` tag pointing to the commit that needs the fix?
Fixes tags help when trying to maintain a stable branch, without a fixes tag a commit could be missed.
Would these two tags make sense?
```
Fixes: 4b383f736e9e ("drivers: caam: implement NXP CAAM Driver - DSA")
Fixes: 503b5c013761 ("drivers: caam: implement NXP CAAM Driver - ECC")
```

**Comment by sahilnxp** [comment]
@jenswi-linaro `Fixes:` tags added in the commit.

File: core/drivers/crypto/caam/acipher/caam_ecc.c [hunk 4]
```diff
@@ -322,6 +322,70 @@ static TEE_Result do_gen_keypair(struct ecc_keypair *key, size_t key_size)
...
+ * because of limitation of Class 2 context register size on i.MX8M series.
+ * So this will be applicable on P384 and P521 ECC curves because these
+ * curves have key_size more than 40bytes.
+ *
+ * @msg_length: Message Length in bytes
+ * @key_size: Key size in bytes
+ */
+static bool padding_required(size_t msg_length, size_t key_size)
+{
+	if (IS_ENABLED(CFG_NXP_CAAM_C2_CTX_REG_WA) && msg_length > 40 &&
```
Comment (jenswi-linaro) [inline core/drivers/crypto/caam/acipher/caam_ecc.c:360]: How about:
``` C
return msg_mes_rep(msg_length, key_size) && msg_length < key_size;
```

It's shorter and also highlights the connection to `msg_mes_rep()`.

By the way, what if msg_length is larger than key_size, is that an error?

File: core/drivers/crypto/caam/acipher/caam_ecc.c [hunk 5]
```diff
@@ -322,6 +322,70 @@ static TEE_Result do_gen_keypair(struct ecc_keypair *key, size_t key_size)
...
+ * will be 48bytes and Message size will also be 48bytes.
+ * This will work only when we set MES_REP = 0 in descriptor,
+ * but in this case, we don't need the padding to be done on the message.
+ *
+ * @msg_length: Message Length in bytes
+ * @key_size: Key size in bytes
+ */
+static bool msg_mes_rep(size_t msg_length, size_t key_size)
+{
+	if (IS_ENABLED(CFG_NXP_CAAM_C2_CTX_REG_WA) && msg_length > 40 &&
```
Comment (jenswi-linaro) [inline core/drivers/crypto/caam/acipher/caam_ecc.c:340]: How about:
``` C
return IS_ENABLED(CFG_NXP_CAAM_C2_CTX_REG_WA) &&
       msg_length > 40 && key_size > 40;
```

File: core/drivers/crypto/caam/acipher/caam_ecc.c (hunk 4 above)
Comment (jenswi-linaro) [reply core/drivers/crypto/caam/acipher/caam_ecc.c:360]: Thanks

**Comment by sahilnxp** [comment]
PR updated.

**Comment by sahilnxp** [comment]
Thanks everyone for quick review. Tags updated.

**Pull request merged**